### ISSUE TITLE
'directives' does not exist in type 'Component'

### ISSUE THREAD

**Issue by pbasnagetr** [NagRock/ngx-auto-scroll#8]
In your example you say to add the directive like so:
```
@Component({
   selector: 'sample',
   templateUrl: 'sample.component.html',
   styleUrls: ['sample.component.css'],
   directives: [NgxAutoScroll]
})
```
doing so gives me the following error:
```Argument of type '{ selector: string; template: any; styles: any[]; directives: (typeof NgxAutoScroll)[]; }' is not assignable to parameter of type 'Component'.
  Object literal may only specify known properties, and 'directives' does not exist in type 'Component'.
```

It seems that the property 'directives' is not valid. I'm using the latest version of Angular (4.4.5), what's the proper way to inject the directive with this restriction?

**Comment by shamskhalil** [comment]
directives has been deprecated in favor of @NgModule.
use : 
`@NgModule({
  declarations: [
    NgxAutoScroll
  ],
...
})`
in app.module.ts

**Comment by beradrian** [comment]
I created a PR to fix this issue. This will imported as a module now.

**Comment by NagRock** [comment]
Fixed in 1.1.0 version. Thanks @beradrian
